File: java/src/dataStructures/Stack.java
package dataStructures;

import java.util.LinkedList;

import util.Printer;

public class Stack<T> {
	private LinkedList<T> list;
	
	public Stack(){
		list = new LinkedList<T>();
	}
	
	public int size(){
		return list.size();
	}
	
	public boolean isEmpty(){
		return list.size() == 0;
	}
	
	public void push(T item){
		list.addFirst(item);
	}
	
	public void pushAll(T...items){
		for (T item : items) {
			push(item);
		}
	}
	
	public T pop(){
		if(isEmpty()) throw new IllegalStateException();
		return list.removeFirst();
	}
	
	public T peak(){
		if(isEmpty()) throw new IllegalStateException();
		return list.getFirst();
	}
	
	public void print(){
		Printer.puts(list.toString());
	}
	
	public static void main(String[] args) {
		Stack<Integer> s = new Stack<Integer>();
		
		s.pushAll(1,2,3,4,5,6,7);
		s.print();
		int n = s.pop();
		Printer.puts(n);
		s.print();
	}
}	

File: js/lib/stack/stack.js
// const LinkedList = require('../linked-list/linked-list');
//
// function Stack(maxSize){
//   this.list = new LinkedList();
//   this.maxSize = maxSize || 20;
// }
//
// Stack.prototype.push = function (value) {
//   if (this.size() >= this.maxSize) {
//     throw new Error('This Stack is full')
//   }
//
//   this.list.addFirst(value);
// };
//
// Stack.prototype.pushAll = function (args) {
//     for (var i = 0; i < arguments.length; i++) {
//       this.push(arguments[i]);
//     }
// };
//
// Stack.prototype.pop = function () {
//   return this.list.removeFirst();
// };
//
// Stack.prototype.peek = function () {
//   return this.list.getFirst();
// };
//
// Stack.prototype.size = function () {
//   var l = this.list;
//   return this.list.length();
// };
//
// Stack.prototype.toString = function () {
//   var s = '-';
//   var i = this.list.iterator()
//   var curr = i.next();
//
//   while(!curr.done){
//       s+= '\n'+ curr.value;
//       curr = i.next();
//   }
//   s += '\n-'
//
//   console.log(s);
//   return s;
// };
//
// if (require.main === module) {
//   var s = new Stack(5);
//   s.pushAll(1,2,3,4,5);
//   s.toString()
//   console.log(s.peek());
//   console.log(s.pop());
//   s.toString();
// }
"use strict";